src/orion/core/worker/consumer.py

```diff
@@ -65,12 +69,12 @@ def consume(self, trial):
         :type trial: `orion.core.worker.trial.Trial`
 
         """
-        log.debug("### Create new directory at '%s':", self.dir)
-        temp_dir = self.experiment.working_dir is None
+        log.debug("### Create new directory at '%s':", self.working_dir)
+        temp_dir = self.working_dir is None
```

Comment:
This will always be True as well.

Reply:
How will this always be true?

Reply:
Always be False, sorry, because `self.working_dir` is always set in the init. Before fixing this, you should add a unit test that detects if not passing --working-dir has the intended effect of using a tmp dir.